## INFO 2950 Final Project: Insert Title Here

## Table of Contents
1. Introduction
    a. Key Terms
    b. Research Questions
    c. Summary of Findings
2. Data Description
3. Preregistration Statements
4. Data Analysis
5. Evaluation of Significance
6. Interpretation and Conclusions
7. Limitations
8. Appendix - Data Cleaning

## Introduction


In [16]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import datetime
import duckdb

## Data Description:

### __MOTIVATION__
__Why was this dataset created?__
- Our dataset was created to observe the factors that influenced life satisfaction in various countries. We wanted to isolate factors that demonstrated the strongest correlation to overall happiness. The five datasets we analyzed all compared individual factors that would influence or measure the wellbeing of the country. Their purpose was to capture the overall state of different countries. 

__Who funded the creation of the dataset?__
- These datasets took information from organizations like the “World Health Organization”, “World bank” statistics, United Nations Statistics, and other datasets. It is unclear who funded these datasets. 
### __Composition__
__Is any information missing from individual instances?__
- There is information missing from individual instances, such as data from smaller countries like Andorra, which may have less resources to collect the data. These rows of missing values have been removed.

__What are the observations (rows) and the attributes (columns)?__

- For these datasets, the rows are the names of countries, and the columns are health, economic, and demography factors. These include number of suicides, infant mortality rate, crime rates, and so on. 

__What processes might have influenced what data was observed and recorded and what was not?__
- The data has mainly been collected from research studies, using methods such as surveys. Thus, there could be several biases that could have influenced data collection, such as selection bias, sampling bias, and social desirability bias.


### __Collection Process__
__How was the data associated with each instance acquired?__
- We gathered the data from various CSV files in order to analyze various facets of influential attributes that impacted a nation. The individual CSV files took data from large health organizations that measured population statistics. Some examples are the "https://ourworldindata.org/", the "Sustainable Development Solutions Network", and the "World Health Organization".

__Over what timeframe was the data collected?__
- Our research analyzed data from the year 2013 to 2023. This data allows us to predict the more recent trends in Happiness levels, GDP, birth rate, etc. 

__What preprocessing was done, and how did the data come to be in the form that you are using?__
- The data scrapped data from large organizations that provided public data. They also took data from previous research studies that used the same data. The data was changed to its current form once we removed less relevant columns and collected the correlation value. Additionally we dropped empty column values to focus on the analyzable data. 

### __Uses__
__Where can your raw source data be found, if applicable? Provide a link to the raw data (hosted on Github, in a Cornell Google Drive or Cornell Box).__

Overall, we used 6 datasets. They can be found in their raw data sources as follows. 

1. Mental health disorders dataset:
https://github.com/yixuanzhou1221/info2950/blob/main/Phase%202/Mental%20health%20Depression%20disorder%20Data.csv

2. Suicide dataset:
https://github.com/yixuanzhou1221/info2950/blob/main/Phase%202/master.csv

3. Demography and economic information dataset:
https://github.com/yixuanzhou1221/info2950/blob/main/Phase%202/world-data-2023.csv

4. Crime information dataset:
https://github.com/yixuanzhou1221/info2950/blob/main/Phase%202/crime-rate-by-country-2023.csv

5. Country and continent dataset:
https://github.com/yixuanzhou1221/info2950/blob/main/Phase%202/countryContinent.csv

6. Happiness information dataset:
https://github.com/yixuanzhou1221/info2950/blob/main/Phase%202/2022.csv

__What (other) tasks could the dataset be used for?__
- These datasets contain a lot of rich information that can be used for a variety of different tasks. For example, other ideas that could be studied include the patterns between education levels and human development, between labor force participation and overall economic performance, and between urbanization trends and environmental consequences.

__Are there tasks for which the dataset should not be used?__
- Some potential tasks that these data should not be used for are visualizing trends over time, because not all the datasets include data over time. Additionally, further verification of the data should be done before using it to inform any policy recommendations.




## Preregistration Statements
**Hypothesis 1**: 
Countries that are reported to have an overall higher rate of happiness will experience lower levels of suicide rates. 
**Analysis**: We will run a linear regression model to determine if there is a relationship between happiness score and suicide rates (number of suicides per 100,000 people).This would be our formula: Happiness Score = α + βsuicide_rate * Suicide Rate + ε. We will be looking to see if βsuicide_rate < 0, to determine if there is a negative correlation between happiness and suicide rates per country.

**Hypothesis 2**: 
Smaller population sizes of countries correspond to lower birth rates, higher minimum wages, and lower crime index, which in turn leads to higher happiness scores for countries.
**Analysis**: We will run a multivariable linear regression to determine the correlation of the inputs of birth rate (number of live births per 1,000 people in the population), minimum wage (in dollars), crime index (out of 100), and population (in number of people) in affecting the output of happiness score. We will test whether 𝛽population< 0 , 𝛽birth_rate< 0, 𝛽min_wage > 0, and 𝛽crime< 0. Our formula would be: Happiness Score = α + βbirth_rate * Birth_rate + βminimum_wage * Minimum wage + βcrime * Crime + βpopulation * Population + ε.

## Data Cleaning:

### Happiness table

First we loaded in the Happiness CSV file

In [17]:
#load the world happiness score of 2022
happiness_df = pd.read_csv('2022.csv', thousands=',')
print(happiness_df.shape)

(147, 12)


The original dataset displays the world happiness country ranking of 2022 and scores from various perspectives. We removed the rows with empty data and are only keeping the happiness rank, happiness score, and country name for further analysis.

In [18]:

#drop the rows where any of these columns is empty.
happiness_df = happiness_df.dropna(subset=['Country', 'Happiness score'])

#select the 'Happiness rank' and 'Country' columns from the dataset
happiness_df = happiness_df[['Country', 'Happiness score']]

print(happiness_df.shape)
happiness_df

(146, 2)


Country  Happiness score
0        Finland           7821.0
1        Denmark           7636.0
2        Iceland           7557.0
3    Switzerland           7512.0
4    Netherlands           7415.0
..           ...              ...
141    Botswana*           3471.0
142      Rwanda*           3268.0
143     Zimbabwe           2995.0
144      Lebanon           2955.0
145  Afghanistan           2404.0

[146 rows x 2 columns]

### Economy and Crime

Note: we are printing dataframe below because there are so many columns that it goes off the page when presented as a table. Printing allows for all the columns of data to be visible. For this project, we will print the dataframes that have columns going off the page. For the other dataframes, we will present them as tables since that provides for easier viewing.

In [19]:
#load the economic data of countries for 2023
econ_df = pd.read_csv('world-data-2023.csv', thousands=',')
print(econ_df.shape)
print(econ_df)

(195, 35)
         Country  Density\n(P/Km2) Abbreviation Agricultural Land( %)  \
0    Afghanistan                60           AF                58.10%   
1        Albania               105           AL                43.10%   
2        Algeria                18           DZ                17.40%   
3        Andorra               164           AD                40.00%   
4         Angola                26           AO                47.50%   
..           ...               ...          ...                   ...   
190    Venezuela                32           VE                24.50%   
191      Vietnam               314           VN                39.30%   
192        Yemen                56           YE                44.60%   
193       Zambia                25           ZM                32.10%   
194     Zimbabwe                38           ZW                41.90%   

     Land Area(Km2)  Armed Forces size  Birth Rate  Calling Code  \
0          652230.0           323000.0       

The original dataset includes basic information about 195 countries with 35 attributes for each country, such as population density, abbreviation, and much more. We are only using information regarding the economy from this dataset, such as GDP, to analyze how economic factors within countries affect citizens’ happiness. We cleaned the rows with missing data and renamed some of the columns to clarify the unit and explain the economic terms. We created a column called ‘GDP Per Capita’ based on the provided GDP and Population information and divided them to gain this essential metric of economics. In addition, we removed special signs such as dollar signs and percentage signs and converted them to numbers for later analysis. We ended up with data for 128 countries.

In [20]:
#calculate the GDP Per Capita 
econ_df['GDP'] = econ_df['GDP'].str.replace(',', '') #remove commas
econ_df['GDP'] = econ_df['GDP'].str.replace('$', '') #remove dollar sign 
econ_df['GDP'] = pd.to_numeric(econ_df['GDP'], errors='coerce') #convert to numeric
econ_df['GDP Per Capita'] = econ_df['GDP'] / econ_df['Population']

econ_df['Minimum wage'] = econ_df['Minimum wage'].str.replace('$', '') #remove dollar sign 
econ_df['Minimum wage'] = pd.to_numeric(econ_df['Minimum wage'], errors='coerce') #convert to numeric

econ_df['Tax revenue (%)'] = econ_df['Tax revenue (%)'].str.replace('%', '') #remove percentage sign
econ_df['Tax revenue (%)'] = pd.to_numeric(econ_df['Tax revenue (%)'], errors='coerce') #convert to numeric
econ_df['Unemployment rate'] = econ_df['Unemployment rate'].str.replace('%', '') #remove percentage sign

#rename the columns for better clarification 
econ_df.rename(columns={'CPI': 'Consumer Price Index'}, inplace=True)
econ_df.rename(columns={'Unemployment rate': 'Unemployment rate(%)'}, inplace=True)
econ_df.rename(columns={'GDP': 'GDP($)'}, inplace=True)
econ_df.rename(columns={'Minimum wage': 'Minimum wage($)'}, inplace=True)

#select the columns related to the economy factor
econ_df = econ_df[['Country', 'GDP($)', 'Population', 'GDP Per Capita',
                   'Minimum wage($)', 'Unemployment rate(%)',
                   'Consumer Price Index', 'Tax revenue (%)']]

#drop the rows where any of these columns is empty.
econ_df = econ_df.dropna(subset=['Country', 'GDP($)', 'Population', 
                                 'GDP Per Capita','Minimum wage($)', 
                                 'Unemployment rate(%)','Consumer Price Index', 
                                 'Tax revenue (%)'])

print(econ_df.shape)
econ_df

(128, 8)


Country        GDP($)   Population  GDP Per Capita  \
0      Afghanistan  1.910135e+10   38041754.0      502.115487   
1          Albania  1.527808e+10    2854191.0     5352.857411   
2          Algeria  1.699882e+11   43053054.0     3948.343279   
4           Angola  9.463542e+10   31825295.0     2973.591160   
6        Argentina  4.496634e+11   44938712.0    10006.148974   
..             ...           ...          ...             ...   
186  United States  2.142770e+13  328239523.0    65280.682241   
187        Uruguay  5.604591e+10    3461734.0    16190.126957   
189        Vanuatu  9.170589e+08     299882.0     3058.065676   
191        Vietnam  2.619212e+11   96462106.0     2715.276036   
193         Zambia  2.306472e+10   17861030.0     1291.343357   

     Minimum wage($) Unemployment rate(%)  Consumer Price Index  \
0               0.43                11.12                149.90   
1               1.12                12.33                119.05   
2               0.95                11.70                151.36   
4               0.71                 6.89                261.73   
6               3.35                 9.79                232.75   
..               ...                  ...                   ...   
186             7.25                14.70                117.24   
187             1.66                 8.73                202.92   
189             1.56                 4.39                117.13   
191             0.73                 2.01                163.52   
193             0.24                11.43                212.31   

     Tax revenue (%)  
0                9.3  
1               18.6  
2               37.2  
4                9.2  
6               10.1  
..               ...  
186              9.6  
187             20.1  
189             17.8  
191             19.1  
193             16.2  

[128 rows x 8 columns]

Next we loaded the crime data for countries during the year 2023

In [21]:
#load the crime data of countries for 2023
crime_df = pd.read_csv('crime-rate-by-country-2023.csv', thousands=',')
print(crime_df.shape)
crime_df

(133, 4)


rank               country  crimeIndex     pop2023
0       1             Venezuela       83.76  28838499.0
1       2      Papua New Guinea       80.79  10329931.0
2       3          South Africa       76.86  60414495.0
3       4           Afghanistan       76.31  42239854.0
4       5              Honduras       74.54  10593798.0
..    ...                   ...         ...         ...
128   129           Switzerland       21.62   8796669.0
129   130                  Oman       20.34   4644384.0
130   131                Taiwan       15.46  23923276.0
131   132  United Arab Emirates       15.23   9516871.0
132   133                 Qatar       12.13   2716391.0

[133 rows x 4 columns]

For the Crime dataset, we are renaming the ‘rank’ column to ‘Crime rank’ to avoid conflict with ‘Happiness rank’ after merging the dataframes. We are also renaming ‘country’ to ‘Country’ so that we can merge these three dataframes based on the ‘Country’ column. Additionally, we are renaming ‘crimeIndex’ to ‘Crime Index’ to keep the first letter capitalized for consistency. We are deleting the ‘pop2023’ column as it will be repetitive information. We are again dropping rows with empty data, but in this case, none of them are dropped.

In [22]:
#rename columns
crime_df.rename(columns={'country': 'Country'}, inplace=True) 
crime_df.rename(columns={'crimeIndex': 'Crime Index'}, inplace=True)
crime_df.rename(columns={'rank': 'Crime rank'}, inplace=True)

#select columns
crime_df=crime_df[['Country','Crime rank','Crime Index']]

#drop the rows where any of these columns is empty.
crime_df=crime_df.dropna(subset=['Country', 'Crime Index','Crime rank'])

print(crime_df.shape)
crime_df

(133, 3)


Country  Crime rank  Crime Index
0               Venezuela           1        83.76
1        Papua New Guinea           2        80.79
2            South Africa           3        76.86
3             Afghanistan           4        76.31
4                Honduras           5        74.54
..                    ...         ...          ...
128           Switzerland         129        21.62
129                  Oman         130        20.34
130                Taiwan         131        15.46
131  United Arab Emirates         132        15.23
132                 Qatar         133        12.13

[133 rows x 3 columns]

After merging the three DataFrames (econ_df, crime_df, and happiness_df) using the 'Country' column with inner joins, we now have data for analysis from 81 countries.

In [23]:
#merge three dataframes into a single data set 
merged_df = pd.merge(econ_df, crime_df, on='Country', how='inner')
merged_df = pd.merge(merged_df, happiness_df, on='Country', how='inner')
print(merged_df.shape)
merged_df

(81, 11)


Country        GDP($)   Population  GDP Per Capita  \
0      Afghanistan  1.910135e+10   38041754.0      502.115487   
1          Albania  1.527808e+10    2854191.0     5352.857411   
2          Algeria  1.699882e+11   43053054.0     3948.343279   
3        Argentina  4.496634e+11   44938712.0    10006.148974   
4          Armenia  1.367280e+10    2957731.0     4622.733493   
..             ...           ...          ...             ...   
76  United Kingdom  2.827113e+12   66834405.0    42300.267126   
77   United States  2.142770e+13  328239523.0    65280.682241   
78         Uruguay  5.604591e+10    3461734.0    16190.126957   
79         Vietnam  2.619212e+11   96462106.0     2715.276036   
80          Zambia  2.306472e+10   17861030.0     1291.343357   

    Minimum wage($) Unemployment rate(%)  Consumer Price Index  \
0              0.43                11.12                149.90   
1              1.12                12.33                119.05   
2              0.95                11.70                151.36   
3              3.35                 9.79                232.75   
4              0.66                16.99                129.18   
..              ...                  ...                   ...   
76            10.13                 3.85                119.62   
77             7.25                14.70                117.24   
78             1.66                 8.73                202.92   
79             0.73                 2.01                163.52   
80             0.24                11.43                212.31   

    Tax revenue (%)  Crime rank  Crime Index  Happiness score  
0               9.3           4        76.31           2404.0  
1              18.6          76        42.53           5199.0  
2              37.2          42        52.03           5122.0  
3              10.1          17        63.82           5967.0  
4              20.9         125        22.79           5399.0  
..              ...         ...          ...              ...  
76             25.5          63        46.07           6943.0  
77              9.6          55        47.81           6977.0  
78             20.1          44        51.73           6474.0  
79             19.1          62        46.19           5485.0  
80             16.2          73        43.62           3760.0  

[81 rows x 11 columns]

### Demography
Here we will import a dataframe that contains information about demography factors across different countries. This will be compared to countries' happiness scores to understand if there are any trends or patterns.

In [24]:
world_data_df = pd.read_csv("world-data-2023.csv", thousands = ",")
print(world_data_df)

         Country  Density\n(P/Km2) Abbreviation Agricultural Land( %)  \
0    Afghanistan                60           AF                58.10%   
1        Albania               105           AL                43.10%   
2        Algeria                18           DZ                17.40%   
3        Andorra               164           AD                40.00%   
4         Angola                26           AO                47.50%   
..           ...               ...          ...                   ...   
190    Venezuela                32           VE                24.50%   
191      Vietnam               314           VN                39.30%   
192        Yemen                56           YE                44.60%   
193       Zambia                25           ZM                32.10%   
194     Zimbabwe                38           ZW                41.90%   

     Land Area(Km2)  Armed Forces size  Birth Rate  Calling Code  \
0          652230.0           323000.0       32.49     

Now we rename our columns of interest to have proper syntax so they are easy to access. The factors we are interested in looking at are birth rate, infant mortality, life expectancy, and maternal mortality ratio.

In [25]:
world_data_df = world_data_df.rename(columns={'Birth Rate': 'birth_rate', 
                'Infant mortality': 'infant_mortality', 
                'Life expectancy': 'life_expectancy', 
                'Maternal mortality ratio': 'maternal_mortality_ratio'})
print(world_data_df)

         Country  Density\n(P/Km2) Abbreviation Agricultural Land( %)  \
0    Afghanistan                60           AF                58.10%   
1        Albania               105           AL                43.10%   
2        Algeria                18           DZ                17.40%   
3        Andorra               164           AD                40.00%   
4         Angola                26           AO                47.50%   
..           ...               ...          ...                   ...   
190    Venezuela                32           VE                24.50%   
191      Vietnam               314           VN                39.30%   
192        Yemen                56           YE                44.60%   
193       Zambia                25           ZM                32.10%   
194     Zimbabwe                38           ZW                41.90%   

     Land Area(Km2)  Armed Forces size  birth_rate  Calling Code  \
0          652230.0           323000.0       32.49     

Here we modify our dataframe to include only the columns that are relevant for our analysis. This results in a cleaner, less cluttered dataframe. Columns with NaN values are dropped, so that there are no missing data values present.

In [26]:
world_data_df = duckdb.sql("SELECT Country, birth_rate, infant_mortality, \
life_expectancy, maternal_mortality_ratio FROM world_data_df").df()
world_data_df = world_data_df.dropna()
world_data_df

Country  birth_rate  infant_mortality  life_expectancy  \
0            Afghanistan       32.49              47.9             64.5   
1                Albania       11.78               7.8             78.5   
2                Algeria       24.28              20.1             76.7   
4                 Angola       40.73              51.6             60.8   
5    Antigua and Barbuda       15.33               5.0             76.9   
..                   ...         ...               ...              ...   
190            Venezuela       17.88              21.4             72.1   
191              Vietnam       16.75              16.5             75.3   
192                Yemen       30.45              42.9             66.1   
193               Zambia       36.19              40.4             63.5   
194             Zimbabwe       30.68              33.9             61.2   

     maternal_mortality_ratio  
0                       638.0  
1                        15.0  
2                       112.0  
4                       241.0  
5                        42.0  
..                        ...  
190                     125.0  
191                      43.0  
192                     164.0  
193                     213.0  
194                     458.0  

[181 rows x 5 columns]

Next, we will refer back to the Happiness Score we used previously

In [27]:
happiness_df

Country  Happiness score
0        Finland           7821.0
1        Denmark           7636.0
2        Iceland           7557.0
3    Switzerland           7512.0
4    Netherlands           7415.0
..           ...              ...
141    Botswana*           3471.0
142      Rwanda*           3268.0
143     Zimbabwe           2995.0
144      Lebanon           2955.0
145  Afghanistan           2404.0

[146 rows x 2 columns]

Here, we join the world data dataframe, which contains relevant demography statistics by country, to the world happiness dataframe, which includes happiness scores by country. This creates a single dataframe with all the relevant information for analysis between demography statistics and happiness.

In [28]:
demography_df = duckdb.sql("SELECT world_data_df.Country, birth_rate, \
infant_mortality, life_expectancy, maternal_mortality_ratio, \
\"Happiness score\" FROM world_data_df INNER JOIN happiness_df \
ON world_data_df.Country = happiness_df.Country").df()
demography_df

Country  birth_rate  infant_mortality  life_expectancy  \
0    Afghanistan       32.49              47.9             64.5   
1        Albania       11.78               7.8             78.5   
2        Algeria       24.28              20.1             76.7   
3        Armenia       13.99              11.0             74.9   
4      Australia       12.60               3.1             82.7   
..           ...         ...               ...              ...   
113   Kazakhstan       21.77               8.8             73.2   
114    Lithuania       10.00               3.3             75.7   
115      Nigeria       37.91              75.7             54.3   
116      Romania        9.60               6.1             75.4   
117         Iran       18.78              12.4             76.5   

     maternal_mortality_ratio  Happiness score  
0                       638.0           2404.0  
1                        15.0           5199.0  
2                       112.0           5122.0  
3                        26.0           5399.0  
4                         6.0           7162.0  
..                        ...              ...  
113                      10.0           6234.0  
114                       8.0           6446.0  
115                     917.0           4552.0  
116                      19.0           6477.0  
117                      16.0           4888.0  

[118 rows x 6 columns]

Additionally, we are adding some regional data, specifically continent and subregion, to help categorize groups of countries.

In [29]:
continent_df = pd.read_csv("countryContinent.csv", encoding='ISO-8859-1')
continent_df

country code_2 code_3  country_code     iso_3166_2 continent  \
0          Afghanistan     AF    AFG             4  ISO 3166-2:AF      Asia   
1        Åland Islands     AX    ALA           248  ISO 3166-2:AX    Europe   
2              Albania     AL    ALB             8  ISO 3166-2:AL    Europe   
3              Algeria     DZ    DZA            12  ISO 3166-2:DZ    Africa   
4       American Samoa     AS    ASM            16  ISO 3166-2:AS   Oceania   
..                 ...    ...    ...           ...            ...       ...   
244  Wallis and Futuna     WF    WLF           876  ISO 3166-2:WF   Oceania   
245     Western Sahara     EH    ESH           732  ISO 3166-2:EH    Africa   
246              Yemen     YE    YEM           887  ISO 3166-2:YE      Asia   
247             Zambia     ZM    ZMB           894  ISO 3166-2:ZM    Africa   
248           Zimbabwe     ZW    ZWE           716  ISO 3166-2:ZW    Africa   

          sub_region  region_code  sub_region_code  
0      Southern Asia        142.0             34.0  
1    Northern Europe        150.0            154.0  
2    Southern Europe        150.0             39.0  
3    Northern Africa          2.0             15.0  
4          Polynesia          9.0             61.0  
..               ...          ...              ...  
244        Polynesia          9.0             61.0  
245  Northern Africa          2.0             15.0  
246     Western Asia        142.0            145.0  
247   Eastern Africa          2.0             14.0  
248   Eastern Africa          2.0             14.0  

[249 rows x 9 columns]

The dataframe is cleaned to only include relevant variables.

In [30]:
continent_df = duckdb.sql("SELECT country, continent, sub_region FROM \
continent_df").df()
continent_df

country continent       sub_region
0          Afghanistan      Asia    Southern Asia
1        Åland Islands    Europe  Northern Europe
2              Albania    Europe  Southern Europe
3              Algeria    Africa  Northern Africa
4       American Samoa   Oceania        Polynesia
..                 ...       ...              ...
244  Wallis and Futuna   Oceania        Polynesia
245     Western Sahara    Africa  Northern Africa
246              Yemen      Asia     Western Asia
247             Zambia    Africa   Eastern Africa
248           Zimbabwe    Africa   Eastern Africa

[249 rows x 3 columns]

Then it is joined to the demography data.

In [31]:
demography_df = duckdb.sql("SELECT continent_df.country, continent, \
sub_region, birth_rate, infant_mortality, life_expectancy, \
maternal_mortality_ratio, \"Happiness score\", FROM continent_df INNER JOIN \
demography_df ON continent_df.country = demography_df.Country").df()
demography_df

country continent                 sub_region  birth_rate  \
0    Afghanistan      Asia              Southern Asia       32.49   
1        Albania    Europe            Southern Europe       11.78   
2        Algeria    Africa            Northern Africa       24.28   
3        Armenia      Asia               Western Asia       13.99   
4      Australia   Oceania  Australia and New Zealand       12.60   
..           ...       ...                        ...         ...   
101      Austria    Europe             Western Europe        9.70   
102      Ecuador  Americas              South America       19.72   
103    Lithuania    Europe            Northern Europe       10.00   
104      Romania    Europe             Eastern Europe        9.60   
105       Zambia    Africa             Eastern Africa       36.19   

     infant_mortality  life_expectancy  maternal_mortality_ratio  \
0                47.9             64.5                     638.0   
1                 7.8             78.5                      15.0   
2                20.1             76.7                     112.0   
3                11.0             74.9                      26.0   
4                 3.1             82.7                       6.0   
..                ...              ...                       ...   
101               2.9             81.6                       5.0   
102              12.2             76.8                      59.0   
103               3.3             75.7                       8.0   
104               6.1             75.4                      19.0   
105              40.4             63.5                     213.0   

     Happiness score  
0             2404.0  
1             5199.0  
2             5122.0  
3             5399.0  
4             7162.0  
..               ...  
101           7163.0  
102           5533.0  
103           6446.0  
104           6477.0  
105           3760.0  

[106 rows x 8 columns]

The data can then be separated into dataframes by continent for further analysis.

In [32]:
asia = demography_df[demography_df['continent'] == 'Asia']
asia

country continent          sub_region  birth_rate  \
0            Afghanistan      Asia       Southern Asia       32.49   
3                Armenia      Asia        Western Asia       13.99   
5                Bahrain      Asia        Western Asia       13.99   
6             Bangladesh      Asia       Southern Asia       18.18   
13              Cambodia      Asia  South-Eastern Asia       22.46   
17                 China      Asia        Eastern Asia       10.90   
21                Cyprus      Asia        Western Asia       10.46   
31               Georgia      Asia        Western Asia       13.47   
39                 India      Asia       Southern Asia       17.86   
40             Indonesia      Asia  South-Eastern Asia       18.07   
41                  Iraq      Asia        Western Asia       29.08   
42                Israel      Asia        Western Asia       20.80   
45                 Japan      Asia        Eastern Asia        7.40   
46                Jordan      Asia        Western Asia       21.98   
47            Kazakhstan      Asia        Central Asia       21.77   
49            Kyrgyzstan      Asia        Central Asia       27.10   
51               Lebanon      Asia        Western Asia       17.55   
53              Malaysia      Asia  South-Eastern Asia       16.75   
58              Mongolia      Asia        Eastern Asia       24.13   
62               Myanmar      Asia  South-Eastern Asia       17.55   
64                 Nepal      Asia       Southern Asia       19.89   
70              Pakistan      Asia       Southern Asia       28.25   
74           Philippines      Asia  South-Eastern Asia       20.55   
77          Saudi Arabia      Asia        Western Asia       17.80   
81             Singapore      Asia  South-Eastern Asia        8.80   
86             Sri Lanka      Asia       Southern Asia       15.83   
89            Tajikistan      Asia        Central Asia       30.76   
90              Thailand      Asia  South-Eastern Asia       10.34   
93                Turkey      Asia        Western Asia       16.03   
96  United Arab Emirates      Asia        Western Asia       10.33   
98            Uzbekistan      Asia        Central Asia       23.30   

    infant_mortality  life_expectancy  maternal_mortality_ratio  \
0               47.9             64.5                     638.0   
3               11.0             74.9                      26.0   
5                6.1             77.2                      14.0   
6               25.1             72.3                     173.0   
13              24.0             69.6                     160.0   
17               7.4             77.0                      29.0   
21               1.9             80.8                       6.0   
31               8.7             73.6                      25.0   
39              29.9             69.4                     145.0   
40              21.1             71.5                     177.0   
41              22.5             70.5                      79.0   
42               3.0             82.8                       3.0   
45               1.8             84.2                       5.0   
46              13.9             74.4                      46.0   
47               8.8             73.2                      10.0   
49              16.9             71.4                      60.0   
51               6.4             78.9                      29.0   
53               6.7             76.0                      29.0   
58              14.0             69.7                      45.0   
62              36.8             66.9                     250.0   
64              26.7             70.5                     186.0   
70              57.2             67.1                     140.0   
74              22.5             71.1                     121.0   
77               6.0             75.0                      17.0   
81               2.3             83.1                       8.0   
86               6.4             7

In [33]:
europe = demography_df[demography_df['continent'] == 'Europe']
europe

country continent       sub_region  birth_rate  \
1                   Albania    Europe  Southern Europe       11.78   
7                   Belgium    Europe   Western Europe       10.30   
9    Bosnia and Herzegovina    Europe  Southern Europe        8.11   
11                 Bulgaria    Europe   Eastern Europe        8.90   
20                  Croatia    Europe  Southern Europe        9.00   
22                  Denmark    Europe  Northern Europe       10.60   
26                  Estonia    Europe  Northern Europe       10.90   
28                  Finland    Europe  Northern Europe        8.60   
29                   France    Europe   Western Europe       11.30   
32                  Germany    Europe   Western Europe        9.50   
34                   Greece    Europe  Southern Europe        8.10   
37                  Hungary    Europe   Eastern Europe        9.60   
38                  Iceland    Europe  Northern Europe       12.00   
43                    Italy    Europe  Southern Europe        7.30   
50                   Latvia    Europe  Northern Europe       10.00   
55                    Malta    Europe  Southern Europe        9.20   
59               Montenegro    Europe  Southern Europe       11.73   
65              Netherlands    Europe   Western Europe        9.70   
69                   Norway    Europe  Northern Europe       10.40   
75                   Poland    Europe   Eastern Europe       10.20   
76                 Portugal    Europe  Southern Europe        8.50   
79                   Serbia    Europe  Southern Europe        9.20   
82                 Slovakia    Europe   Eastern Europe       10.60   
83                 Slovenia    Europe  Southern Europe        9.40   
85                    Spain    Europe  Southern Europe        7.90   
87                   Sweden    Europe  Northern Europe       11.40   
88              Switzerland    Europe   Western Europe       10.00   
95                  Ukraine    Europe   Eastern Europe        8.70   
101                 Austria    Europe   Western Europe        9.70   
103               Lithuania    Europe  Northern Europe       10.00   
104                 Romania    Europe   Eastern Europe        9.60   

     infant_mortality  life_expectancy  maternal_mortality_ratio  \
1                 7.8             78.5                      15.0   
7                 2.9             81.6                       5.0   
9                 5.0             77.3                      10.0   
11                5.9             74.9                      10.0   
20                4.0             78.1                       8.0   
22                3.6             81.0                       4.0   
26                2.1             78.2                       9.0   
28                1.4             81.7                       3.0   
29                3.4             82.5                       8.0   
32                3.1             80.9                       7.0   
34                3.6             81.3                       3.0   
37                3.6             75.8                      12.0   
38                1.5             82.7                       4.0   
43                2.6             82.9                       2.0   
50                3.3             74.7                      19.0   
55                6.1             82.3                       6.0   
59                2.3             76.8                       6.0   
65                3.3             81.8                       5.0   
69                2.1             82.8                       2.0   
75                3.8             77.6                       2.0   
76                3.1             81.3                       8.0   
79                4.8             75.5                      12.0   
82                4.6             77.2                       5.0   
83                1.7             81.0                       7.0   
85                2.5             83.3                       4.0   
87         

In [34]:
africa = demography_df[demography_df['continent'] == 'Africa']
africa

country continent       sub_region  birth_rate  infant_mortality  \
2         Algeria    Africa  Northern Africa       24.28              20.1   
8           Benin    Africa   Western Africa       36.22              60.5   
12   Burkina Faso    Africa   Western Africa       37.93              49.0   
14       Cameroon    Africa    Middle Africa       35.39              50.6   
24          Egypt    Africa  Northern Africa       26.38              18.1   
27       Ethiopia    Africa   Eastern Africa       32.34              39.1   
30          Gabon    Africa    Middle Africa       31.61              32.7   
33          Ghana    Africa   Western Africa       29.41              34.9   
35         Guinea    Africa   Western Africa       36.36              64.9   
48          Kenya    Africa   Eastern Africa       28.75              30.6   
52         Malawi    Africa   Eastern Africa       34.12              35.3   
54           Mali    Africa   Western Africa       41.54              62.0   
56      Mauritius    Africa   Eastern Africa       10.20              13.6   
60        Morocco    Africa  Northern Africa       18.94              19.2   
61     Mozambique    Africa   Eastern Africa       37.52              54.0   
63        Namibia    Africa  Southern Africa       28.64              29.0   
68        Nigeria    Africa   Western Africa       37.91              75.7   
78        Senegal    Africa   Western Africa       34.52              31.8   
80   Sierra Leone    Africa   Western Africa       33.41              78.5   
84   South Africa    Africa  Southern Africa       20.51              28.5   
91           Togo    Africa   Western Africa       33.11              47.4   
92        Tunisia    Africa  Northern Africa       17.56              14.6   
94         Uganda    Africa   Eastern Africa       38.14              33.8   
99       Zimbabwe    Africa   Eastern Africa       30.68              33.9   
105        Zambia    Africa   Eastern Africa       36.19              40.4   

     life_expectancy  maternal_mortality_ratio  Happiness score  
2               76.7                     112.0           5122.0  
8               61.5                     397.0           4623.0  
12              61.2                     320.0           4670.0  
14              58.9                     529.0           5048.0  
24              71.8                      37.0           4288.0  
27              66.2                     401.0           4241.0  
30              66.2                     252.0           4958.0  
33              63.8                     308.0           4872.0  
35              61.2                     576.0           4891.0  
48              66.3                     342.0           4543.0  
52              63.8                     349.0           3750.0  
54              58.9                     562.0           4479.0  
56              74.4                      61.0           6071.0  
60              76.5                      70.0           5060.0  
61              60.2                     289.0           5048.0  
63              63.4                     195.0           4459.0  
68              54.3                     917.0           4552.0  
78              67.7                     315.0           5046.0  
80              54.3                    1120.0           3574.0  
84              63.9                     119.0           5194.0  
91              60.8                     396.0           4112.0  
92              76.5                      43.0           4516.0  
94              63.0                     375.0           4603.0  
99              61.2                     458.0           2995.0  
105             63.5                     213.0           3760.0

In [35]:
oceania = demography_df[demography_df['continent'] == 'Oceania']
oceania

country continent                 sub_region  birth_rate  \
4     Australia   Oceania  Australia and New Zealand       12.60   
66  New Zealand   Oceania  Australia and New Zealand       11.98   

    infant_mortality  life_expectancy  maternal_mortality_ratio  \
4                3.1             82.7                       6.0   
66               4.7             81.9                       9.0   

    Happiness score  
4            7162.0  
66           7200.0

In [36]:
americas = demography_df[demography_df['continent'] == 'Americas']
americas

country continent        sub_region  birth_rate  \
10               Brazil  Americas     South America       13.92   
15               Canada  Americas  Northern America       10.10   
16                Chile  Americas     South America       12.43   
18             Colombia  Americas     South America       14.88   
19           Costa Rica  Americas   Central America       13.97   
23   Dominican Republic  Americas         Caribbean       19.51   
25          El Salvador  Americas   Central America       18.25   
36             Honduras  Americas   Central America       21.60   
44              Jamaica  Americas         Caribbean       16.10   
57               Mexico  Americas   Central America       17.60   
67            Nicaragua  Americas   Central America       20.64   
71               Panama  Americas   Central America       18.98   
72             Paraguay  Americas     South America       20.57   
73                 Peru  Americas     South America       17.95   
97              Uruguay  Americas     South America       13.86   
100           Argentina  Americas     South America       17.02   
102             Ecuador  Americas     South America       19.72   

     infant_mortality  life_expectancy  maternal_mortality_ratio  \
10               12.8             75.7                      60.0   
15                4.3             81.9                      10.0   
16                6.2             80.0                      13.0   
18               12.2             77.1                      83.0   
19                7.6             80.1                      27.0   
23               24.1             73.9                      95.0   
25               11.8             73.1                      46.0   
36               15.1             75.1                      65.0   
44               12.4             74.4                      80.0   
57               11.0             75.0                      33.0   
67               15.7             74.3                      98.0   
71               13.1             78.3                      52.0   
72               17.2             74.1                     129.0   
73               11.1             76.5                      88.0   
97                6.4             77.8                      17.0   
100               8.8             76.5                      39.0   
102              12.2             76.8                      59.0   

     Happiness score  
10            6293.0  
15            7025.0  
16            6172.0  
18            5781.0  
19            6582.0  
23            5737.0  
25            6120.0  
36            6022.0  
44            5850.0  
57            6128.0  
67            6165.0  
71            6309.0  
72            5578.0  
73            5559.0  
97            6474.0  
100           5967.0  
102           5533.0

### Health

First we load in the CSV files analyze suicide rates and mental health disorders in various countries

In [37]:
suicide_df = pd.read_csv('master.csv')
mental_health_df = pd.read_csv('Mental health Depression disorder Data.csv', dtype='unicode')
# https://stackoverflow.com/questions/24251219/pandas-read-csv-low-memory-and-dtype-options

#### Mental Disorders
We begin by removing unnecessary columns from the data frame

In [38]:
mental_health_df.drop('Code', axis=1, inplace=True)

We rename the columns to make them more intuitive. We renamed the "Entity" column to the "Country". Then we removed the missing values from the dataframe to ensure we only focus on the relevant data.

In [39]:
#Rename the Entity column to countries
mental_health_df = mental_health_df.rename(columns={'Entity': 'Country'})

# #Remove NaN
mental_health_df = mental_health_df.dropna()

The "Year" column wasn't an int, so we changed it to make it easier to remove and select years later on. We deleted years before 2013 since we wanted to collect more recent data. 

In [40]:
# Make years into int, since it was an object before
mental_health_df['Year'] = mental_health_df['Year'].astype(int)

# remove years before 2013 
mental_health_df = mental_health_df[mental_health_df['Year'] >= 2013]

The disorder columns contained errors in its data that needed to be removed. For example, one of its data contained 2 periods instead 1 for percentages. To solve this we converted all the disorders to numeric values and set all non-passing vales as NaN. We then dropped those values.

In [41]:
# Remove the data in the percentages that aren't numeric. Ex) 
columns_to_convert = [
    'Schizophrenia (%)', 'Bipolar disorder (%)', 'Eating disorders (%)',
    'Anxiety disorders (%)', 'Drug use disorders (%)',
    'Depression (%)', 'Alcohol use disorders (%)']

# change to numeric, ignore errors
# https://stackoverflow.com/questions/44522741/pandas-mean-typeerror-could-not-convert-to-numeric
mental_health_df[columns_to_convert] = mental_health_df[columns_to_convert].apply(pd.to_numeric, 
                                                                                  errors='coerce')
mental_health_df.dropna(subset=columns_to_convert, inplace=True)

We then grouped the data by country and took the mean of the disorder percentages. We dropped the columns that were not relevant to the final dataframe.

In [42]:
# Group the disorder percentage mean for 2013-2017
averages_by_country = mental_health_df.groupby('Country').mean()
averages_by_country.drop(axis=0, columns=['Year','index'], inplace=True)
averages_by_country.reset_index(inplace=True)
averages_by_country

Country  Schizophrenia (%)  Bipolar disorder (%)  \
0                   Afghanistan           0.165882              0.707279   
1                       Albania           0.200605              0.704141   
2                       Algeria           0.197709              0.817824   
3                American Samoa           0.248964              0.468421   
4          Andean Latin America           0.200701              0.881123   
..                          ...                ...                   ...   
226  Western Sub-Saharan Africa           0.174619              0.636718   
227                       World           0.254415              0.596277   
228                       Yemen           0.172382              0.735498   
229                      Zambia           0.165007              0.605270   
230                    Zimbabwe           0.156640              0.609488   

     Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
0                0.104365               4.875703                2.533182   
1                0.170857               3.387972                0.512789   
2                0.211327               5.058440                1.708282   
3                0.179106               3.316180                0.769744   
4                0.350026               4.266526                0.666658   
..                    ...                    ...                     ...   
226              0.139599               2.904105                0.483082   
227              0.207662               3.774304                0.930513   
228              0.136447               4.852630                1.400195   
229              0.138857               3.585710                0.507067   
230              0.120138               3.099608                0.770384   

     Depression (%)  Alcohol use disorders (%)  
0          4.135129                   0.662045  
1          2.205231                   1.847211  
2          3.639566                   0.665544  
3          2.937838                   1.141303  
4          2.606462                   1.548944  
..              ...                        ...  
226        3.757981                   0.967807  
227        3.439976                   1.412415  
228        4.017016                   0.635910  
229        3.581924                   1.632803  
230        3.158223                   1.514015  

[231 rows x 8 columns]

#### Suicide Table

First we renamed the countries column to Country to better align with the other CSV file names. Then we dropped the years before the year 2013 and reset the index for more accurate data collection. 

In [43]:
#Rename the Entity column to countries
suicide_df = suicide_df.rename(columns={'country': 'Country'})

# drop years before 2013
suicide_df = suicide_df[suicide_df['year'] >= 2013]
suicide_df.reset_index(drop=True, inplace=True)

First we added together the suicide numbers if the year and the country was the same. We did this to account for the fact that the data took data based on different age groups. Then, we collected the suicide rates for each country and retook the sum of the suicide number.

In [44]:
# Add the suicide number together if the country and year is the same
suicide_df = suicide_df.groupby(['Country', 'year'])['suicides_no'].sum().reset_index()

# Total number of suicides on the most recent years
suicide_2013_to_2017_df = suicide_df.groupby('Country')['suicides_no'].sum().reset_index()
suicide_2013_to_2017_df

Country  suicides_no
0   Antigua and Barbuda            1
1             Argentina         9291
2               Armenia          266
3             Australia         8526
4               Austria         5057
..                  ...          ...
79              Ukraine        15542
80       United Kingdom        14522
81        United States       128101
82              Uruguay         1767
83           Uzbekistan         4045

[84 rows x 2 columns]

Next we will refer back to the Happiness score we had before

In [45]:
happiness_df

Country  Happiness score
0        Finland           7821.0
1        Denmark           7636.0
2        Iceland           7557.0
3    Switzerland           7512.0
4    Netherlands           7415.0
..           ...              ...
141    Botswana*           3471.0
142      Rwanda*           3268.0
143     Zimbabwe           2995.0
144      Lebanon           2955.0
145  Afghanistan           2404.0

[146 rows x 2 columns]

#### Merge all 3 tables by country

We merged together the suicide, mental disorder, and happiness score dateframe.

In [46]:
merged_1_2_df= pd.merge(averages_by_country, suicide_2013_to_2017_df, on='Country', how='inner')
health_df = pd.merge(merged_1_2_df, happiness_df, on='Country', how='inner')
health_df

Country  Schizophrenia (%)  Bipolar disorder (%)  \
0        Argentina           0.197961              0.772578   
1          Armenia           0.196526              0.716817   
2        Australia           0.364150              1.144993   
3          Austria           0.256640              0.939808   
4          Bahrain           0.206873              0.836296   
..             ...                ...                   ...   
57         Ukraine           0.190860              0.687293   
58  United Kingdom           0.258671              1.086026   
59   United States           0.335195              0.651874   
60         Uruguay           0.198341              0.780724   
61      Uzbekistan           0.187263              0.668441   

    Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
0               0.397766               6.278569                0.989317   
1               0.164855               2.590217                0.495973   
2               0.942223               6.595998                2.181237   
3               0.674277               5.345074                0.909465   
4               0.269325               4.657096                1.710888   
..                   ...                    ...                     ...   
57              0.171669               2.941524                0.696986   
58              0.542953               4.656162                1.696709   
59              0.521523               6.651867                3.272024   
60              0.392434               6.309361                0.933896   
61              0.148843               2.556248                0.477325   

    Depression (%)  Alcohol use disorders (%)  suicides_no  Happiness score  
0         3.657941                   1.823537         9291           5967.0  
1         2.740602                   1.979112          266           5399.0  
2         4.694260                   1.509925         8526           7162.0  
3         3.265730                   1.879006         5057           7163.0  
4         3.896715                   0.731456           14           6647.0  
..             ...                        ...          ...              ...  
57        4.091572                   4.816161        15542           5084.0  
58        4.123517                   1.911172        14522           6943.0  
59        4.799864                   2.020849       128101           6977.0  
60        3.585823                   1.181403         1767           6474.0  
61        2.884805                   1.917761         4045           6063.0  

[62 rows x 10 columns]

## Data Limitations


__1.__ Here are some limitations of the health data:

  - The dataset from the Mental Health Disorder CSV contained dates from 1990 to 2017. Some countries like Albania only contained statistics up to the year 2015. Since the data was not up to date, it could cause inaccuracies in our dataset, making it difficult to make predictions on life satisfaction. 

  - The Mental Health Disorder CSV contained multiple NaN values on different years of different countries. Since there was missing data, our final output could be skewed depending on the years missing. 

  - The Suicide Number CSV could have accuracy problems through the collection process. It is hard to account for all the suicides globally since it doesn’t account for missing people, doesn’t account for underreportings, and we don’t know what constitutes as a suicide. 

  - When merging the different data frames, it was difficult to keep all the countries shared by the three data sets. Since we were inner joining, only countries that were on all datasets were included. 

__2.__ Though the global country information dataset is very comprehensive and has over 18,000 downloads, there isn’t clear documentation of its sources and how the data was collected. Thus, there is some ambiguity associated with reporting findings based on this data.

__3.__ The world happiness report data is not up to date to 2023, so we have chosen to use the most recent data collected from 2022 in our analysis. This could potentially alter our findings as it may not completely and accurately reflect the current situation.

__4.__ The world happiness report data was mainly collected through a survey, the Gallup World Survey. Thus, there could be selection biases in the responses and the data may not be completely representative of the whole population.

__5.__ The Crime Rate By Country 2023 CSV doesn’t clearly explain how the 'crime index' is calculated, which raises concerns about potential bias in data collection.

## Questions For Reviewers:



__1.__ We are unsure of how to find the “weight” of each factor in influencing happiness. Could you provide any insights as to how to approach this?

__2.__ We are interested in possibly doing prediction models using factors we are analyzing as inputs. However, we haven’t included time-series data for some of these factors. Is it still possible to do a prediction model, or should we pivot to a different question?

__3.__ For future analyses, will we need to derive a single equation that combines demographic, health, economic, and crime factors?
